packages/core/ui-settings/core-ui-settings-server-internal/src/clients/ui_settings_user_client.ts

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import type { KibanaRequest } from '@kbn/core-http-server';
+import type { IUserUiSettingsClient } from '@kbn/core-ui-settings-server/src';
+import type { UiSettingsServiceOptions } from '../types';
+import { BaseUiSettingsClient } from './base_ui_settings_client';
+
+interface UserProvidedValue<T = unknown> {
+  userValue?: T;
+  isOverridden?: boolean;
+}
+
+type UserProvided<T = unknown> = Record<string, UserProvidedValue<T>>;
+
+/**
+ * Common logic for setting / removing keys in a {@link IUserUiSettingsClient} implementation
+ */
+export class UiSettingsUserClient extends BaseUiSettingsClient implements IUserUiSettingsClient {
+  private readonly userProfileSettingsClient: UiSettingsServiceOptions['userProfileSettingsClient'];
+  private readonly request?: KibanaRequest;
+
+  constructor(options: UiSettingsServiceOptions, request?: KibanaRequest) {
+    super(options);
+    const { userProfileSettingsClient } = options;
+    this.userProfileSettingsClient = userProfileSettingsClient;
+    this.request = request;
+  }
+
+  async getUserProfileSettings(): Promise<Record<string, string>> {
+    let result = {} as Record<string, string>;
+
+    if (this.request) {
+      result = (await this.userProfileSettingsClient?.get(this.request)) || {};
+    } else {
+      this.log.warn('Request not set, unable to retrieve User Settings');
+    }
+
+    return result;
+  }
+
```

Comment:
Is the intention to create one SO per user to hold other User Settings? What's the benefit we'd get from doing so instead of storing every User Setting in the user's profile? Any known limitations?

Reply:
I don't think there are any limitations for for simple settings, like Dark Mode, being stored in the User Profile, but Im sure there could arise a scenario that wouldn't work.

In my initial meeting with the core team and discussions on my brainstorming doc, the team expressed that there might be a need to store User Settings in SO eventually, or at least, be able to migrate the settings from User Profiles to SO in the future.

I wanted the client to behave like the regular/global client so I included the methods stubbed so they would fulfill the interface and be available if we do need to move that direction